torch/onnx/_onnx_supported_ops.py

```diff
@@ -75,32 +74,49 @@ def _symbolic_argument_count(func):
     return params
 
 
-def _all_symbolics_schemas():
-    symbolics_schemas: Dict[str, _TorchSchema] = {}
+def _all_symbolics_schemas() -> Dict[str, _TorchSchema]:
+    symbolics_schemas = {}
+
+    for name in registration.registry.all_functions():
+        func_group = registration.registry.get_function_group(name)
+        assert func_group is not None
+        symbolics_schema = _TorchSchema(name)
+        func = func_group.get(_constants.onnx_main_opset)
+        if func is not None:
+            symbolics_schema.arguments = _symbolic_argument_count(
+                func_group.get(_constants.onnx_main_opset)
+            )
+            symbolics_schema.opsets = list(
+                range(func_group.get_min_supported(), _constants.onnx_main_opset + 1)
+            )
+        else:
+            # Only support opset < 9
+            func = func_group.get(7)
+            symbolics_schema.opsets = list(range(7, _constants.ONNX_BASE_OPSET))
+
+        symbolics_schemas[name] = symbolics_schema
 
-    for domain, version in symbolic_registry._registry:
-        for opname, sym_func in symbolic_registry._registry[(domain, version)].items():
-            symbolics_schema = _TorchSchema("aten::" + opname)
-            symbolics_schema.arguments = _symbolic_argument_count(sym_func)
-            if opname in symbolics_schemas:
-                symbolics_schemas[opname].opsets.append(version)
-            else:
-                symbolics_schema.opsets = [version]
-                symbolics_schemas[opname] = symbolics_schema
     return symbolics_schemas
 
 
 def onnx_supported_ops():
-    aten_schemas = _all_aten_forward_schemas()
+    all_schemas = _all_forward_schemas()
     symbolic_schemas = _all_symbolics_schemas()
     torch_schemas = set(symbolic_schemas.values())
     supported_ops = []
     onnx_supported = []
-    for schema in aten_schemas:
+    for schema in all_schemas:
         if schema in torch_schemas:
-            opname = schema.name[6:]  # without "aten::" prefix
+            opname = schema.name
             opsets = symbolic_schemas[opname].opsets
             if schema not in supported_ops:
                 supported_ops.append(symbolic_schemas[opname])
-                onnx_supported.append((opname, " ".join(str(o) for o in opsets)))
+                onnx_supported.append(
+                    (
+                        opname,
+                        f"{opsets[0]}-{opsets[-1]}"
+                        if len(opsets) > 1
+                        else f"{opsets[0]}",
+                    )
+                )
```

Comment:
Is this file all about annotation and formatting? If that's the case, I feel like it'd better we keep only the described features in that certain PR, and open another for related formatting.

Reply:
SG

Reply:
Done